### PR TITLE
Add Unregister Push Token Request

### PR DESCRIPTION
**Pull request by kennyklaviyo**
# Description

# Check List

- [x] Are you changing anything with the public API?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you tested this change on real device?
- [x] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

### CODE REVIEW

File: sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequestTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                      "type": "profile",
+                      "attributes": {
+                        "email": "$EMAIL",
+                        "phone_number": "$PHONE",
+                        "external_id": "$EXTERNAL_ID",
+                        "anonymous_id": "$ANON_ID"
+                      }
+                    }
+                  },
+                  "device_metadata": {
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequestTest.kt:58]: Oh actually, does unregister need device metadata?

Comment (kennyklaviyo) [reply sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequestTest.kt:58]: haha yeah i was just removing this

File: sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package com.klaviyo.analytics.networking.requests
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt:1]: NVM, i think without the metadata there will be very little duplication i think

File: sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    PLATFORM to DeviceProperties.platform,
+                    VENDOR to VENDOR_FCM
+                )
+            )
+        )
+    }
+
+    override fun equals(other: Any?): Boolean {
+        return when (other) {
+            is UnregisterPushTokenApiRequest -> body.toString() == other.body.toString()
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt:53]: Ooh, will company ID need to be accounted for here? Same would go for the register call.

Comment (evan-masseau) [reply sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt:53]: Although, I guess anonymous ID should always be present and different if company has changed.

**Review by evan-masseau — APPROVED** [review]
Looks good, I think we can test that question about company ID in the next phase.

File: sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt (hunk 3 above)
Comment (kennyklaviyo) [reply sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/UnregisterPushTokenApiRequest.kt:53]: ill merge this so i can test with your changes but i think we'll just end up removing this entirely, no need to dedup unregister i think

**Pull request merged**